models/refugee.py

```diff
@@ -0,0 +1,118 @@
+        :param name of refugee family
+        :param number of family member
+        :param camp of the refugee family locating at
+        :param medical condition of the refugee
+        :param start date of refugee family
+        :param closing date of refugee family
+        """
+
+        self.name = self.__checkAndFormatName(firstname, lastname)
+        self.numOfFamilyMember = self.__checkNumOfFamilyMember(numOfFamilyMember)
+        self.camp = camp
+        self.medicalCondition = medicalCondition
+        self.dateOfCreation = datetime.today().date()
+        self.dateOfClosing = self.__checkDateOfClosing(dateOfClosing)
+
+
+    def __checkAndFormatName(self, firstname, lastname):
+        """
+        format name to a new name
+        """
+        if not isinstance(firstname, str) or not isinstance(lastname, str):
+            raise self.InvalidNameException()
+        elif not firstname.isalpha() or not lastname.isalpha():
+            raise self.InvalidNameException()
+        else:
+            fullname = firstname + " " + lastname
+            return fullname
+        #error handling if invalid name is inputed
+
+    def __checkNumOfFamilyMember(self, numOfFamilyMember):
+        """
+        format number of family member
+        """
+        if not isinstance(numOfFamilyMember, int) or numOfFamilyMember < 0:
+            raise self.InvalidNumOfFamilyMemberException()
+        return numOfFamilyMember + 1
+
+    def formatCamp(self, newCamp):
+        """
+        format identification camp of refugee
+        """
+        if newCamp not in camp.Camp.camps:
+            raise self.InvalidCampException(newCamp)
+        self.camp = newCamp
+
+    def formatMedicalCondition(self, medicalCondition): #medical condition types provided or a description?
+        """
+        format refugee family medical condition
+        """
+        self.medicalCondition = medicalCondition
+
+    def __checkDateOfClosing(self, dateOfClosing: date): #for removing and archive
+        """
+        format closing date of refugee family
+        """
+        today = datetime.today().date()
+        if today >= dateOfClosing:
+            raise self.InvalidClosingDateException()
+        elif dateOfClosing == None:
+            return dateOfClosing
+        else:
+            return dateOfClosing
+
+
+    def __str__(self):
+        return f"Refugee family {self.name} located in {self.camp}.\n"\
+               f"number of family member: {self.numOfFamilyMember}\n" \
+               f"medical condition: {self.medicalCondition}\n" \
+               f"creation date: {self.dateOfCreation}\n" \
+               f"closing date: {self.dateOfClosing}\n"
+
+
+    class InvalidNumOfFamilyMemberException(Exception):
+        """
+        Raise exception when number of family member entered.
+        """
+        def __init__(self):
+            super().__init__(f"Invalid number of family member. Family member must be an positive integer.")
```

Comment:
Would it be possible to change the message to say:

"Invalid input: the number of family members must be a positive integer."